test/e2e/app-dir/dynamic-io-request-apis/dynamic-io-request-apis.test.ts

```diff
@@ -0,0 +1,126 @@
+import { nextTestSetup } from 'e2e-utils'
+
+const WITH_PPR = !!process.env.__NEXT_EXPERIMENTAL_PPR
+
+const stackStart = /\s+at /
+
+function createExpectError(cliOutput: string) {
+  let cliIndex = 0
+  return function expectError(
+    containing: string,
+    withStackContaining?: string
+  ) {
+    const initialCliIndex = cliIndex
+    let lines = cliOutput.slice(cliIndex).split('\n')
+
+    let i = 0
+    while (i < lines.length) {
+      let line = lines[i++] + '\n'
+      cliIndex += line.length
+      if (line.includes(containing)) {
+        if (typeof withStackContaining !== 'string') {
+          return
+        } else {
+          while (i < lines.length) {
+            let stackLine = lines[i++] + '\n'
+            if (!stackStart.test(stackLine)) {
+              expect(stackLine).toContain(withStackContaining)
+            }
+            if (stackLine.includes(withStackContaining)) {
+              return
+            }
+          }
+        }
+      }
+    }
+
+    expect(cliOutput.slice(initialCliIndex)).toContain(containing)
+  }
+}
+
+describe(`Request Promises`, () => {
+  describe('On Prerender Completion', () => {
+    const { next, isNextDev, skipped } = nextTestSetup({
+      files: __dirname + '/fixtures/reject-hanging-promises-static',
+      skipStart: true,
+    })
+
+    if (skipped) {
+      return
+    }
+
+    if (isNextDev) {
```

Comment:
`test/e2e` will run on dev/start/deploy -- maybe this is a better test for `test/production` if we only want it to run on `start`?

I'm not sure if this is expected to work on deployment since forked PRs won't currently run those tests but my guess is that we only want to look at build output and not deploy anything.

Reply:
As long as it doesn't fail the deployment tests, sgtm!